src/code/actions.js

```diff
@@ -146,12 +149,12 @@ function restrictToIntegerRange(value, minValue, maxValue) {
 export function navigateToCurrentRoute(routeSpec) {
   const {level, mission, challenge} = routeSpec;
   return (dispatch, getState) => {
-    const { authoring: levels } = getState(),
-          levelCount = levels.length,
+    const authoring = getState().authoring,
+          levelCount = AuthoringUtils.getLevelCount(authoring),
           nextLevel = restrictToIntegerRange(level, 0, levelCount - 1),
-          missionCount = levels[nextLevel].length,
+          missionCount = AuthoringUtils.getMissionCount(authoring, nextLevel),
           nextMission = restrictToIntegerRange(mission, 0, missionCount - 1),
-          challengeCount = levels[nextLevel][nextMission].length,
+          challengeCount = AuthoringUtils.getChallengeCount(authoring, nextLevel, nextMission);
```

Comment:
Sorry, I pushed an unfinished version, I should have unassigned in the meantime. It works now!